File: frontend/src/test/mockAnalysis.ts
import type { AnalysisResult } from '../types/api'

export const mockAnalysis: AnalysisResult = {
  summary: 'The TypeScript build stops on a strict type mismatch.',
  root_cause: 'An optional environment value is passed to a required string parameter.',
  confidence: 0.94,
  evidence: [
    {
      source: 'ci.log',
      line_start: 8,
      line_end: 9,
      excerpt: "Type 'undefined' is not assignable to type 'string'.",
      explanation: 'The compiler identifies the nullable argument.',
    },
  ],
  relevant_files: ['src/services/report.ts'],
  recommended_changes: ['Validate REPORT_BUCKET at startup.'],
  patch: 'diff --git a/src/config/env.ts b/src/config/env.ts',
  verification_commands: ['npm run typecheck'],
  risks: ['Deployments must define the variable.'],
  missing_information: ['Deployment variables were not supplied.'],
  analysis_time_ms: 1_324,
  compression_stats: {
    available: true,
    paritok_connected: true,
    hosted_gpu_available: true,
    verification: 'local_health+hosted_gpu_preflight+stats_delta',
    proxy_version: '1.0.0',
    model: 'deepseek-v4-flash',
    proxy_requests: 1,
    original_tokens: 8_000,
    compressed_tokens: 2_000,
    saved_tokens: 6_000,
    compression_ratio: 0.25,
    cumulative: {
      total_requests: 12,
      input_tokens_original: 80_000,
      input_tokens_compressed: 20_000,
      compression_ratio: 0.25,
      tokens_saved: 60_000,
      tools_filtered: 0,
    },
    cost_estimate: {
      estimated_input_cost_saved_usd: 0.00084,
      input_cache_miss_usd_per_m_tokens: 0.14,
      pricing_snapshot_date: '2026-07-31',
      disclaimer: "Estimate from LeanCI's configured DeepSeek price; not an actual bill.",
    },
    message:
      "Verified through Paritok; Token metrics come only from this request's stats delta.",
  },
}

File: backend/app/mock_analysis.py
"""Deterministic application analysis data.

This module does not import, configure, or call Paritok, DeepSeek, or any other
external service.
"""

from app.models import AnalysisResult, EvidenceItem, UnavailableCompressionStats


def build_mock_analysis() -> AnalysisResult:
    """Return a stable UI-development contract with unavailable token metrics."""

    return AnalysisResult(
        summary=(
            "The TypeScript build stops while compiling the report service because "
            "an optional environment value is assigned to a required string."
        ),
        root_cause=(
            "`process.env.REPORT_BUCKET` has the type `string | undefined`, but "
            "`uploadReport` requires a definite string. Strict null checks correctly "
            "reject the call before the bundle is produced."
        ),
        confidence=0.94,
        evidence=[
            EvidenceItem(
                source="ci.log",
                line_start=8,
                line_end=9,
                excerpt=(
                    "src/services/report.ts(42,19): error TS2345: Argument of type "
                    "'string | undefined' is not assignable to parameter of type 'string'."
                ),
                explanation=(
                    "The compiler points to the upload destination argument and identifies "
                    "the exact nullable type mismatch."
                ),
            ),
            EvidenceItem(
                source="src/services/report.ts",
                line_start=42,
                line_end=42,
                excerpt="await uploadReport(process.env.REPORT_BUCKET, payload)",
                explanation=(
                    "The environment variable is read at the call site without validation "
                    "or a fallback."
                ),
            ),
        ],
        relevant_files=[
            "src/services/report.ts",
            "src/config/env.ts",
            ".github/workflows/ci.yml",
        ],
        recommended_changes=[
            "Validate REPORT_BUCKET once during application startup.",
            "Pass the validated configuration value into the report service.",
            "Add a regression test for the missing-variable failure path.",
        ],
        patch=(
            "diff --git a/src/config/env.ts b/src/config/env.ts\n"
            "index 132fa11..8be1c22 100644\n"
            "--- a/src/config/env.ts\n"
            "+++ b/src/config/env.ts\n"
            "@@ -1,3 +1,8 @@\n"
            "+const reportBucket = process.env.REPORT_BUCKET\n"
            "+\n"
            "+if (!reportBucket) {\n"
            "+  throw new Error('REPORT_BUCKET is required')\n"
            "+}\n"
            "+\n"
            "+export { reportBucket }\n"
        ),
        verification_commands=[
            "npm run typecheck",
            "npm test -- src/config/env.test.ts",
            "npm run build",
        ],
        risks=[
            "Failing at startup changes when the configuration error becomes visible.",
            "Existing deployment environments must define REPORT_BUCKET.",
        ],
        missing_information=[
            "The current deployment environment variable inventory was not provided.",
            "No report service unit test was included with this demo log.",
        ],
        compression_stats=UnavailableCompressionStats(),
        analysis_time_ms=0,
    )
